qa/tasks/cephfs/test_volumes.py

```diff
@@ -155,6 +156,57 @@ def test_subvolume_create_with_desired_data_pool_layout_in_group(self):
         desired_pool = self.mount_a.getfattr(subvol2_path, "ceph.dir.layout.pool")
         self.assertEqual(desired_pool, new_pool)
 
+    def test_subvolumegroup_create_with_desired_mode(self):
+        group1 = self._generate_random_group_name()
+        group2 = self._generate_random_group_name()
+        # default mode
+        expected_mode1 = "755"
+        # desired mode
+        expected_mode2 = "777"
+
+        # create group
+        self._fs_cmd("subvolumegroup", "create", self.volname, group1)
+        self._fs_cmd("subvolumegroup", "create", self.volname, group2, "--mode", "777")
+
+        group1_path = os.path.join('volumes', group1)
+        group2_path = os.path.join('volumes', group2)
+
+        # check group's mode
+        actual_mode1 = self.mount_a.run_shell(['stat', '-c' '%a', group1_path]).stdout.getvalue().strip()
+        actual_mode2 = self.mount_a.run_shell(['stat', '-c' '%a', group2_path]).stdout.getvalue().strip()
+        self.assertEqual(actual_mode1, expected_mode1)
+        self.assertEqual(actual_mode2, expected_mode2)
+
+    def test_subvolume_create_with_desired_mode_in_group(self):
+        subvol1 = self._generate_random_subvolume_name()
+        subvol2 = self._generate_random_subvolume_name()
+        subvol3 = self._generate_random_subvolume_name()
+        group = self._generate_random_group_name()
+        # default mode
+        expected_mode1 = "755"
+        # desired mode
+        expected_mode2 = "777"
+
+        # create group
+        self._fs_cmd("subvolumegroup", "create", self.volname, group)
+
+        # create subvolume in group
+        self._fs_cmd("subvolume", "create", self.volname, subvol1, "--group_name", group)
+        self._fs_cmd("subvolume", "create", self.volname, subvol2, "--group_name", group, "--mode", "777")
+        # check whether mode 0777 also works
+        self._fs_cmd("subvolume", "create", self.volname, subvol3, "--group_name", group, "--mode", "0777")
+        subvol1_path = os.path.join('volumes', group, subvol1)
```

Comment:
why not use "subvolume getpath" command?

Reply:
Maybe? I think it's okay to construct the path for a unit test.

Reply:
It's a pretty minor change, +1 if it can be done now.

Reply:
Done